easybuild/framework/easyblock.py

```diff
@@ -2202,8 +2202,30 @@ def make_module_step(self, fake=False):
                 for line in txt.split('\n'):
                     self.dry_run_msg(' ' * 4 + line)
         else:
-            write_file(mod_filepath, txt)
-            self.log.info("Module file %s written: %s", mod_filepath, txt)
+            module_only = build_option('module_only')
+            backup_modules = build_option('backup_modules')
+            if module_only and os.path.exists(mod_filepath) and backup_modules and not fake:
+                warning_msg = "Old module file found. Backing it up in %s."
+                hidden = False
+                if isinstance(self.module_generator, ModuleGeneratorTcl):
+                     hidden = True
```

Comment:
Because lua modules have to end in `.lua`, otherwise they aren't treated as modules by Lmod, and don't show up with `ml av`. So there is no need to keep them hidden since the backup extension avoids that file ends in `.lua`.

Tcl modules on the other need to be hidden. Which is not that nice, since Lmod can still show them with `--show-hidden`. But that's the best compromise IMO. The alternative would be to remove the module header.

Backing up outside of the module tree kind of defeats the purpose of the backup. In my opinion the backup should be used for quick verification of the changes and for doing a "rollback" if necessary. If I have the current module and the backup in different files it slightly makes it more complicated and cumbersome, and more difficult to directly see if there was a previous version at any point with a simple `ls`